packages/openscd/src/open-scd.ts

```
@@ -278,15 +259,51 @@ export class OpenSCD extends LitElement {
     if (src.startsWith('blob:')) URL.revokeObjectURL(src);
   }
 
+  /**
+   *
+   * @deprecated Use `handleConfigurationPluginEvent` instead
+   */
+  public handleAddExternalPlugin(e: AddExternalPluginEvent){
+    this.addExternalPlugin(e.detail.plugin);
+    const {name, kind} = e.detail.plugin
+
+    const event = newConfigurePluginEvent(name,kind, e.detail.plugin)
+
+    this.handleConfigurationPluginEvent(event)
+  }
+
+
+  public handleConfigurationPluginEvent(e: ConfigurePluginEvent){
+    const { name, kind, config } = e.detail;
+
+    const hasPlugin = this.hasPlugin(name, kind);
+    const hasConfig = config !== null;
+    const isChangeEvent = hasPlugin && hasConfig
+    const isRemoveEvent = hasPlugin && !hasConfig
+    const isAddEvent = !hasPlugin && hasConfig
+
+    // the `&& config`is only because typescript
+    // cannot infer that `isChangeEvent` and `isAddEvent` implies `config !== null`
+    if(isChangeEvent && config){
+      this.changePlugin(config)
+
+    }else if( isRemoveEvent ){
+      this.removePlugin(name, kind)
+
+    }else if(isAddEvent && config){
+      this.addPlugin(config)
+
+    }else{
+      console.error("Invalid plugin configuration event", {name, kind, config})
```

For logging we should use the log events, so the user can the error message

---

This is meant for developers, should the users see this?

---

If this is something that should never happen unless something is wrong then I would throw an error